src/components/Progressbar.tsx

```diff
@@ -0,0 +1,103 @@
+import classNames from 'classnames';
+import { FC, PropsWithChildren } from 'react';
+
+type Color = 'gray' | 'blue' | 'red' | 'green' | 'yellow' | 'indigo' | 'purple';
+type Size = 'sm' | 'md' | 'lg' | 'xl';
+
+export type ProgressbarProps = PropsWithChildren<{
+  className?: string;
+  progress?: number;
+  color?: Color;
+  size?: Size;
+  label?: string;
+  labelPosition?: 'inside' | 'outside';
+  labelProgress?: boolean;
+  labelSize?: Size;
+  labelColor?: Color;
+}>;
+
+const colorClasses: Record<Color, string> = {
+  blue: 'bg-blue-600',
+  red: 'bg-red-600',
+  purple: 'bg-purple-600',
+  gray: 'bg-gray-600',
+  green: 'bg-green-600',
+  yellow: 'bg-yellow-400',
+  indigo: 'bg-indigo-600',
+};
+const sizeClasses: Record<Size, string> = {
+  sm: 'h-1.5',
+  md: 'h-2.5',
+  lg: 'h-4',
+  xl: 'h-6',
+};
+const labelSizeClasses: Record<Size, string> = {
+  sm: 'text-xs',
+  md: 'text-md',
+  lg: 'text-lg',
+  xl: 'text-xl',
+};
+const lableColorClasses: Record<Color, string> = {
+  blue: 'text-blue-600',
+  red: 'text-red-600',
+  purple: 'text-purple-600',
+  gray: 'text-gray-600',
+  green: 'text-green-600',
+  yellow: 'text-yellow-400',
+  indigo: 'text-indigo-600',
+};
+export const Progressbar: FC<ProgressbarProps> = ({
+  className,
+  progress = 45,
+  color = 'blue',
+  size = 'md',
+  label,
+  labelPosition = 'inside',
+  labelProgress = false,
+  labelSize = 'md',
+  labelColor = 'blue',
+}) => {
+  return (
+    <>
+      {label && labelPosition === 'outside' && (
+        <div className="mb-1 flex justify-between">
+          <span
+            className={classNames(
+              'font-medium dark:text-white',
+              labelSizeClasses[labelSize],
+              lableColorClasses[labelColor],
+            )}
+          >
+            {label}
+          </span>
+          {labelProgress && (
+            <span
+              className={classNames(
+                'font-medium dark:text-white',
+                labelSizeClasses[labelSize],
+                lableColorClasses[labelColor],
```

Comment:
we can move all calls of `lableColorClasses` and `lableSizeClasses` to the parent suggested `div` to reduce the redundancy in the code.